Source/Core/VideoCommon/Assets/MeshAsset.cpp

```diff
@@ -0,0 +1,597 @@
+      }
+    }
+
+    // Position matrix can be enabled if the draw that is using
+    // this mesh needs it
+    chunk.vertex_declaration.posmtx.enable = false;
+
+    data->m_mesh_chunks.push_back(std::move(chunk));
+  }
+
+  return true;
+}
+
+bool ReadGLTFNodes(std::string_view mesh_file, const tinygltf::Model& model,
+                   const tinygltf::Node& node, const Common::Matrix44& mat, MeshData* data)
+{
+  if (node.mesh != -1)
+  {
+    if (!ReadGLTFMesh(mesh_file, model, model.meshes[node.mesh], mat, data))
+      return false;
+  }
+
+  for (std::size_t i = 0; i < node.children.size(); i++)
+  {
+    const tinygltf::Node& child = model.nodes[node.children[i]];
+    const auto child_mat = mat * BuildMatrixFromNode(child);
+    if (!ReadGLTFNodes(mesh_file, model, child, child_mat, data))
+      return false;
+  }
+
+  return true;
+}
+
+bool ReadGLTFMaterials(std::string_view mesh_file, const tinygltf::Model& model, MeshData* data)
+{
+  for (std::size_t i = 0; i < model.materials.size(); i++)
+  {
+    const tinygltf::Material& material = model.materials[i];
+
+    // TODO: support converting material data into Dolphin material assets
+    data->m_mesh_material_to_material_asset_id.insert_or_assign(material.name, "");
+  }
+
+  return true;
+}
+
+bool ReadGLTF(std::string_view mesh_file, const tinygltf::Model& model, MeshData* data)
+{
+  int scene_index = model.defaultScene;
+  if (scene_index == -1)
+    scene_index = 0;
+
+  const auto& scene = model.scenes[scene_index];
+  const auto scene_node_indices = scene.nodes;
+  for (std::size_t i = 0; i < scene_node_indices.size(); i++)
+  {
+    const tinygltf::Node& node = model.nodes[scene_node_indices[i]];
+    const auto mat = BuildMatrixFromNode(node);
+    if (!ReadGLTFNodes(mesh_file, model, node, mat, data))
+      return false;
+  }
+
+  return ReadGLTFMaterials(mesh_file, model, data);
+}
+}  // namespace
+bool MeshData::FromJson(const VideoCommon::CustomAssetLibrary::AssetID& asset_id,
+                        const picojson::object& json, MeshData* data)
+{
+  if (const auto iter = json.find("material_mapping"); iter != json.end())
+  {
+    if (!iter->second.is<picojson::object>())
+    {
+      ERROR_LOG_FMT(
+          VIDEO,
+          "Asset '{}' failed to parse json, expected 'material_mapping' to be of type object",
+          asset_id);
+      return false;
+    }
+
+    for (const auto& [material_name, asset_id_json] : iter->second.get<picojson::object>())
+    {
+      if (!asset_id_json.is<std::string>())
+      {
+        ERROR_LOG_FMT(
+            VIDEO,
+            "Asset '{}' failed to parse json, material name '{}' linked to a non-string value",
+            asset_id, material_name);
+        return false;
+      }
+
+      data->m_mesh_material_to_material_asset_id[material_name] = asset_id_json.to_str();
+    }
+  }
+  return true;
+}
+
+void MeshData::ToJson(picojson::object* obj, const MeshData& data)
```

Comment:
```suggestion
void MeshData::ToJson(picojson::object& obj, const MeshData& data)
```

which gets rid of the check and makes the code much more straightforward

Reply:
Thanks.  Thought I had fixed them all :)